08_flask_html_basics.md

```diff
@@ -0,0 +1,592 @@
+
+### Exercise 8.12
+
+Although you haven't explicitly mentioned it in the code, while communicating with the `/` route of your application through the browser, one of the HTTP methods has to be used. Which of the above is most likely to be used?
+
+<details markdown="1"><summary>Hint</summary>
+While the Flask application is running, you can refresh the page and check the terminal in VS Code. The method used, along with the path requested, should appear in the log.
+</details>
+
+After you've figured out the HTTP method, try setting it explicitly in the code. You can do that by adding a `methods` parameter to the app routing, like this:
+
+```python
+@app.route("/", methods=['HTTP_METHOD'])
+```
+
+Refresh the page to check that everything is still working.
+
+### HTTP Response Codes
+
+Since HTTP is a stateless, request-response protocol, the server needs to communicate to the client the status of the request (e.g. whether it was successful or not), in order for the client to interpret the server's response correctly.
+
+HTTP uses response codes to communicate this. They are composed of three digits, 1xx-5xx, where the first digit signifies the class of responses:
+- 1xx informational response - the request was received by the server
+- 2xx successful - the request was processed successfully
+- 3xx redirect - the client is redirected to another resource or path
+- 4xx client error - the client's request is invalid
+- 5xx server error - there is an error with the server, when responding to an apparently correct request
+
+### Exercise 8.13
+
+Refresh the page and check out Flask's output in the VS Code terminal.
+
+- What status code is displayed for the `/` endpoint?
+- What status code is displayed for the image?
+- Why is the image status code different from the page's? You can check out [here](https://www.rfc-editor.org/rfc/rfc9110.html#name-304-not-modified) an explanation for the status code that you should be seeing.
+
+Try changing the HTTP operation in your code and refresh the page. What status code are you seeing now and why?
+
+<details markdown="1"><summary>Click here to reveal the answers</summary>
+
+- The status code for the `/` route should be `200` (OK), signifying that the request completed successfully
+- The status code for the image should be `200` for the first request, but upon refreshing the page it should be `304` (Not Modified). The client already has a copy of the image, so the server instructs the client to use the local copy instead of downloading a new one.
+- If you change the HTTP method in your endpoint to anything else but `GET`, you should receive a `405` (Method not allowed) response when refreshing the page.
+
+</details>
+
+
+### Exercise 8.14
+
+You will need to add a new endpoint to your application, mapping a route called `/add`. This will be responsible for adding an animal name to the existing list. The method should have a single parameter, which you can assume to be string.
```

Comment:
```suggestion
You will need to add a new endpoint to your application, mapping a route called `/add`. This will be responsible for adding an animal name to the existing list. The method should have a single parameter, which you can assume to be a string.
```